src/poetry/console/commands/show.py

```diff
@@ -65,6 +81,23 @@ def handle(self) -> int | None:
         if self.option("tree"):
             self.init_styles(self.io)
 
+        if self.option("why"):
+            if self.option("tree") and package is None:
+                self.line_error(
+                    "<error>Error: --why requires a package when combined with"
+                    " --tree.</error>"
+                )
+
+                return 1
+
+            if not self.option("tree") and package:
+                self.line_error(
+                    "<error>Error: --why cannot be used when displaying a single"
```

Comment:
Maybe this could be phrased as "--why cannot be used without --tree when displaying a single package"?